lib/fizzy/execute.cpp

```diff
@@ -833,16 +855,21 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
             assert(cur_pages <= size_t(std::numeric_limits<int32_t>::max()));
             const auto new_pages = cur_pages + delta;
             assert(new_pages >= cur_pages);
-            uint32_t ret = static_cast<uint32_t>(cur_pages);
-            try
-            {
-                if (new_pages > instance.memory_pages_limit)
-                    throw std::bad_alloc();
-                memory->resize(new_pages * PageSize);
-            }
-            catch (std::bad_alloc const&)
-            {
+            uint32_t ret;
```

Comment:
Hmm, wondering if the following would be more readable:
```
uint32_t = static_cast<uint32_t>(-1);
if (new_pages <= instance.memory_pages_limit)
{
  try
  {
    memory->resize(new_pages * PageSize);
    ret = static_cast<uint32_t>(cur_pages);
  }
  catch ...
}
```

Reply:
It looks fine, I don't mind.

The best is to wrap this into `uint32_t Memory::grow(something);`. Then you eliminate `ret` housekeeping.

Reply:
I can move it into a simple helper for now.